### ISSUE TITLE
S3 - Attacker can recover sensitive information from a previous version of a file stored on S3.

### ISSUE THREAD

**Issue by msaindane** [owasp-cloud-security/owasp-cloud-security#53]
By default, on a versioned S3 bucket, AWS does not delete previous versions of the file automatically.

Threat: Information Disclosure

Mitigations could include:
* Remove all versions of sensitive files instead of only deleting the latest version.
* Backup sensitive files before permanently deleting them.
* Enable strict access controls on S3 buckets

**Comment by zeroXten** [comment]
Is this because of the use of delete markers?

**Comment by msaindane** [comment]
Yes, when a user deletes a file, AWS just adds a delete marker and previous versions of the files can still be accessed. In the UI, unless you disable "Hide" versions, you do not see the file. This could give a false sense of security to the user.

Ideally, if a user wants to delete a file containing sensitive data, they should delete all the versions of the file. Unless there is a business case to keep a particular version in AWS.

This is one of the problems with versioned buckets.

**Comment by zeroXten** [comment]
This is a good one. I would imagine a lot of people get caught out by this.
